### DISCUSSION TITLE
How to convert open source docker to pro docker without changing the secret key

### DISCUSSION THREAD

**Discussion by tuandinh91** [rustdesk/rustdesk-server-pro#353] [Q&A] [answered]
Hi,
As the title said, I plan to upgrade my open source rustdesk docker to pro version but I search the guide for a while but no hope. Can you help me to give the guide to convert the open source server to pro server please? The current server has about 800 devices already, so I would like to keep the same secret key please.
Thank you very much
Ted

**Comment by rustdesk** [comment]
https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/#migrate-to-new-host-and-backup--restore

**Reply by tuandinh91** [reply]
Hi , I think you misunderstood me. I would like to convert current OS server that was installed to Docker to PRO Server while keep the secret key so the current Devices can still be able to join the server with out changing the secret key. Can you show me how to achieve it? As long as I got the guide, I will be ready to purchase the license for 40 user and 1500 devices, thank you.

**Reply by rustdesk** [reply, marked as answer]
The `id_ed25519`file  https://github.com/rustdesk/rustdesk-server-pro/discussions/184#discussioncomment-8013375
